### ISSUE TITLE
Change default compile/transpile target to be `es2019`

### ISSUE THREAD

**Issue by sidoruk-sv** [wix/yoshi#2594]
# 🙋 Feature Request

Originally initiated under #2586

### 😯 Current Behavior

For now, our projects are compiled by typescript to `"target": "es5"`

### 🤔 Expected Behavior

Due to the fact that we usually have polyfill.io bundle available on Wix pages. We could upgrade the compile target to `es2019`.

### 🔦 Context 

Original comment from @ranyitz:
> Regarding babel-preset-yoshi. We would like to configure it according to the platform it will be used in.
>
> This is surely a possible optimization, but it is a possible breaking change of production so we need to do this very carefully.
>
> Please open a dedicated issue to discuss it. Hopefully, we'll be able to do that for TypeScript as well.

**Comment by AdirAmsalem** [comment]
I like the idea however I think I'm missing something so I'd love to join the discussion.
Targeting `es2019`, how are we going to support the syntax itself which we cannot polyfill and is not supported natively - things like `class`, `... (spread)`, `=> (arrow functions)`, `template literals`, etc? We have to transpile it.

**Comment by just-jeb** [comment]
According to our [convo](https://wix.slack.com/archives/C0NJPS8N5/p1589958546476500) the limiting factor is Safary 12. 
We can start by checking the highest supported ECMA script version (syntax wise) and consider targeting to this version.  
Another important thing that must be taken into account is that most likely such a change will require `polyfills.io` query modification on each Wix page, which, as every cross-Wix effort, is not that trivial.

**Comment by sidoruk-sv** [comment]
Hi @AdirAmsalem thanks for the questions)

1. We have a list of supported browsers:
https://support.wix.com/en/article/supported-browsers

2. About new syntax:
2.1. `class` https://caniuse.com/#feat=es6-class
2.2. `...spread` https://caniuse.com/#search=JavaScript%20operator%3A%20spread
2.3. `(arrow) => 'functions'` https://caniuse.com/#feat=arrow-functions
2.4. `template literal` https://caniuse.com/#feat=template-literals

As you could mention all that functionality is supported by browsers that we support at Wix.
And pros of generating and bundling the whole amount of helpers and extra code seems like less that cons for amount of traffic for all of that redundant code.

Also migrating to new targets from `es5` could bring more easily tree-shaking and other staff
> need help here from the experts on `bundling "the new code" VS bundling "support all browsers" code` maybe @yavorsky could help here)

The only exception here is Viewer and UserOfUser context due to support of IE (should be targeted by platform in Yoshi), and only there the old target could remain as "support as much as we can".

**Comment by AdirAmsalem** [comment]
I was referring specifically to UoU flows where supporting IE11 is indeed the issue.

The issues with transpiling user-facing code to es2019 and uou-facing code to es5 is:
1. It's error-prone - unless it's 100% controlled by the infra.
2. Any shared cross-segment library will have to be transpiled to es5, because (currently) we do not transpile dependencies. Probably all libraries.

**Comment by ranyitz** [comment]
I think that our Lowest hanging fruit is the editor & business-manager flows, there we know what is the target of our code. In business-manager it's always the same and for editor, it would be different for editor-app, settings, and viewer-app. @amitdahan @yavorsky  

I agree with @AdirAmsalem that we can't do this without knowing the target for sure and we'll face a dangerous breaking change.

Another option for the rest of the apps would be to specify browserlist configuration, we can then create [sharable configs](https://github.com/browserslist/browserslist#shareable-configs) for each target and transpile according to it. (Should work perfectly with babel) It's worth saying that we are considering to transpile the ts code using babel, because it's faster and because we can have more control. @ronami

**Comment by ranyitz** [comment]
@sidoruk-sv I also agree with @AdirAmsalem that talking about polyfills and transpilation target on the same issue can be confusing. I prefer focusing on this discussion on the transpilation target and I will appreciate if you'll edit the issue description to reflect that.

**Comment by ranyitz** [comment]
Regarding libraries, I would start with transpiling them to es5. In the future we'll be able to maybe create 2 targets for a lib, but we're not there yet. I would focus this discussion about apps.
